front/app/containers/PasswordReset/PasswordReset.test.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { screen, render, fireEvent, waitFor } from 'utils/testUtils/rtl';
+
+jest.mock('utils/cl-router/Link', () => 'Link');
+
+jest.mock('utils/cl-router/history', () => ({
+  location: {
+    search: '?token=test-token',
+  },
+  push: jest.fn(),
+}));
+
+jest.mock('services/auth', () => ({
+  resetPassword: () => jest.fn(),
+}));
+
+import PasswordReset from './index';
+
+describe('PasswordReset', () => {
+  it('renders a success message with a login button after password reset', async () => {
+    render(<PasswordReset />);
+
+    fireEvent.change(screen.getByLabelText(/password/i), {
+      target: {
+        value: 'test-new-password',
+      },
+    });
+
+    const submitButton = screen.getByText(/Confirm new password/i);
+    fireEvent.click(submitButton);
+
+    await waitFor(() => {
+      expect(
+        screen.getByText('Your password has been successfully updated.')
+      ).toBeInTheDocument();
+      expect(screen.getByRole('button')).toHaveTextContent('Log in');
+    });
+  });
```

Comment:
A few things:
-We should use userEvent rather than fireEvent
-Try to use byRole unless it's not possible

Check the links I shared during the last CDM. :-) So then we'd get:

```
  it('renders a success message with a login button after password reset', async () => {
    const user = userEvent.setup();
    render(<PasswordReset />);

    // Enter new password
    const passwordInput = screen.getByLabelText(/password/i);
    await user.type(passwordInput, 'test-new-password');
    // Submit
    const submitButton = screen.getByRole('button', {
      name: /confirm new password/i,
    });
    await user.click(submitButton);

    // Test if it succeeded
    const successMessage = await screen.findByText(
      'Your password has been successfully updated.'
    );
    const loginButton = await screen.findByRole('button', {
      name: /log in/i,
    });
    expect(successMessage).toBeInTheDocument();
    expect(loginButton).toBeInTheDocument();
  });
```

Reply:
Did we decide on what to use between userEvent rather than fireEvent? I remember the discussion being that Iva would find out and communicate ideal approach. Happy to use either one

Reply:
Iva was going to check it for herself indeed. RTL [recommends userEvent](https://testing-library.com/docs/dom-testing-library/api-events).

Reply:
Happy to change to userEvent in this case since it is recommend and it works. We can revisit when that changes